src/Compilers/Core/Portable/Operations/IIndexOperation.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    /// <summary>
+    /// Represents an index operation.
+    /// <para>
+    /// Current Usage:
+    ///  (1) C# index expressions
+    /// </para>
+    /// </summary>
+    /// <remarks>
+    /// This interface is reserved for implementation by its associated APIs. We reserve the right to change it in the future.
+    /// </remarks>
+    public interface IIndexOperation : IOperation
+    {
+        /// <summary>
+        /// The operand.
+        /// </summary>
+        IOperation Operand { get; }
+
+        /// <summary>
+        /// <code>true</code> if this is a 'lifted' index operation.  When there is an 
+        /// operator that is defined to work on a value type, 'lifted' operators are 
+        /// created to work on the <see cref="System.Nullable{T}"/> versions of those
+        /// value types.
+        /// </summary>
+        bool IsLifted { get; }
+
+        /// <summary>
+        /// Constructor symbol that's called to create this Index value.
```

Comment:
>Constructor symbol [](start = 12, length = 18)

I don't understand how this relates to indexing. What does indexing have to do with a constructor? Can you please provide an example?

Reply:
I don't think we should be using a constructor to create the "from end" values. In order for the API to work reasonably from other languages, the values should probably be created using a static factory.

---
In reply to: [206305990](https://github.com/dotnet/roslyn/pull/28259#discussion_r206305990) [](ancestors = 206305990)